airbyte-cdk/java/airbyte-cdk/db-sources/src/test/java/io/airbyte/cdk/integrations/source/relationaldb/state/CursorStateMessageProducerTest.java

```diff
@@ -224,41 +225,36 @@ void testIteratorCatchesExceptionWhenEmissionFrequencyNonZero() {
     // frequency minimum of 1 record
     assertEquals(createStateMessage(RECORD_VALUE_2, 2, 4.0), iterator.next());
     // no further records to read since Exception was caught above and marked iterator as endOfData()
-    assertFalse(iterator.hasNext());
+    assertThrows(RuntimeException.class, () -> iterator.hasNext());
```

Comment:
Wait what? How is it acceptable for `hasNext` to throw an exception? I had to look in the `SourceStateIterator` code to understand why. Can you define a new exception class which extends `RuntimeException` and assert on that instead?

Reply:
done for adding a new exception.

idea we have for all state iterators is just to throw if the underlying record iterator has exception. Since it's going to be resumable platform will do retries.